src/ips/graph_utils.rs

```diff
@@ -79,7 +80,23 @@ pub fn construct_graph(nodes: &[Node]) -> Graph<SocketAddr> {
 
     for node in nodes {
         let node_addr = node.addr;
+
+        // This is a hack to add nodes that are not connected to any other node.
+        // This is needed to run some graph algorithms on the graph - like counting betweenness or
+        // closeness centrality as well as simple getting degree.
+        if node.connections.is_empty() {
+            graph.insert(Edge::new(node_addr, node_addr));
+            continue;
+        }
+
         for i in &node.connections {
+            if *i >= nodes.len() {
+                println!(
+                    "Node {} has connection to non-existing node {}",
```

Comment:
when is this the case? because it sounds wrong - so maybe we can leave a comment with an example or something